ocaml/xapi-cli-server/record_util.ml

```diff
@@ -706,6 +706,25 @@ let host_sched_gran_to_string = function
   | `socket ->
       "socket"
 
+let host_numa_affinity_policy_to_string = function
+  | `any ->
+      "any"
+  | `besteffort ->
+      "best-effort"
+  | `default ->
+      "default"
+
+let host_numa_affinity_policy_of_string = function
+  | "any" ->
+      `any
+  | "best-effort" ->
+      `besteffort
+  | "default" ->
+      `default
+  | s ->
+      raise
+        (Record_failure ("Expected 'any', 'best-effort' or 'default', got " ^ s))
```

Comment:
That is a problem everywhere in this file, and would more easily be fixed if the file was auto-generated, I'll take a look whether that is easy to do.

Reply:
We're not very consistent here, sometimes we use `_` as separator for the CLI, sometimes `-`, and the actual polymorphic variant sometimes includes neither (e.g. `balanceslb is `balance-slb` on the CLI). We could autogenerate code to accept both _ and - as separator, but we need to be backwards compatible with the way we printed things before (and changing the enum name to add _ would break some SDKs that got generated without it).

Still we can do better than what we do now by having a list of polymorphic variant and string names and calling a helper function to perform the matching and print the error message.